jax/experimental/mosaic/gpu/core.py

```diff
@@ -546,13 +546,20 @@ def _initialize_scratch(
 def _declare_runtime_functions():
   """Declares the runtime functions that can be used by the generated code."""
   ptr_ty = ir.Type.parse("!llvm.ptr")
+  i32 = ir.IntegerType.get_signless(32)
   i64 = ir.IntegerType.get_signless(64)
   arg_tys = [ptr_ty, ptr_ty, i64, i64, ptr_ty, ptr_ty, i64, ptr_ty]
   init_tma_desc_type = ir.FunctionType.get(arg_tys, [])
   func.FuncOp(
       "mosaic_gpu_init_tma_desc", init_tma_desc_type, visibility="private"
   )
-
+  rtype = ir.IntegerType.get_signless(32)
+  ftype = ir.Type.parse(f'!llvm.func<{rtype} ()>')
```

Comment:
nit: just use the `"!llvm.func<i32()>"` literal. There's no need to format in the type if it's constant. Same below with `ptr_ty`. Either get the types in Python and build the signature using `FunctionType.get` or use string literals without referring to the Python types